kml.cc

```diff
@@ -406,34 +406,28 @@ void wpt_icon(xg_string args, const QXmlStreamAttributes*)
 
 void trk_coord(xg_string args, const QXmlStreamAttributes*)
 {
-  int consumed = 0;
-  double lat, lon, alt;
-  Waypoint* trkpt;
-  int n = 0;
-
   route_head* trk_head = route_head_alloc();
-
-  QString iargs = args;
   if (wpt_tmp && !wpt_tmp->shortname.isEmpty()) {
     trk_head->rte_name  = wpt_tmp->shortname;
   }
   track_add_head(trk_head);
-  while ((n = sscanf(CSTRc(iargs), "%lf,%lf,%lf%n", &lon, &lat, &alt, &consumed)) > 0) {
-    trkpt = new Waypoint;
-    trkpt->latitude = lat;
-    trkpt->longitude = lon;
-
-    // Line malformed or two-arg format without alt .  Rescan.
-    if (2 == n) {
-      sscanf(CSTRc(iargs), "%lf,%lf%n", &lon, &lat, &consumed);
-    }
 
-    if (3 == n) {
-      trkpt->altitude = alt;
-    }
+  auto vecs = args.simplified().split(' ');
+  for(const auto& vec : vecs) {
```

Comment:
I recommend running clazy when you touch any Qt container, especially in a range based for loop:
qmake QMAKE_CXX=/path/to/clazy; make
You may want to export a modified CLAZY_CHECKS variable, but clazy-range-loop is probably enabled by default as it is included in the level 1 checks.

> kml.cc:416:3: warning: c++11 range-loop might detach Qt container (QStringList) [-Wclazy-range-loop]
  for(const auto& vec : vecs) {
  ^
                        qAsConst( )

The clazy suggested fix will work, but an alternative I prefer when possible is to make the container const:
`const auto vecs = args.simplified.split(' ');`

Reply:
Integrated. Thanx. It drives me crazy that 'auto' doesn't always return the "best" type.

Should we work on a clazy pass as a presubmit test?